Sources/Brave/WebFilters/ResourceDownloaderStream.swift

```diff
@@ -50,21 +50,34 @@ struct ResourceDownloaderStream: Sendable, AsyncSequence, AsyncIteratorProtocol
       try await Task.sleep(seconds: fetchInterval)
       
       do {
-        let result = try await resourceDownloader.download(resource: resource)
-        
-        switch result {
-        case .downloaded(let url, let date):
-          return .success(DownloadResult(date: date, fileURL: url))
-        case .notModified:
-          continue
-        }
-      } catch {
+        let result = try await downloadResult()
+        guard result.isModified else { continue }
+        return .success(result)
+      } catch let error as URLError {
+        // Soft fail these errors
         return .failure(error)
+      } catch {
+        throw error
       }
     }
   }
   
+  private func downloadResult() async throws -> DownloadResult {
+    switch try await resourceDownloader.download(resource: resource) {
+    case .downloaded(let url, let date):
+      return DownloadResult(date: date, fileURL: url, isModified: true)
+    case .notModified(let url, let date):
+      return DownloadResult(date: date, fileURL: url, isModified: false)
+    }
+  }
+  
   func makeAsyncIterator() -> ResourceDownloaderStream {
     return self
   }
+  
+  static func downloadResult(for resource: Resource) throws -> DownloadResult? {
```

Comment:
As talked about before, this name is super confusing. `DownloadedResult` would be better as a type, `result` would be better for the name of the method

Reply:
If you don't mind id like to do this in the proposed cleanup above. Because changing this will result in several places in code changing. And technically its a cached result so what about `cachedResult`?

Reply:
I did change the method name to `cachedResult`

Reply:
I don't see this changed

Reply:
Sorry I changed only one of the methods. But just so you know this method actually downloads. I was thinking you were talking about the other method which returns cached data. This one actually does the download. So I made it `result()`

Reply:
I didn't change DownloadResult yet. Not exactly sure what you want me to change `DownloadResult` to